plugins/modules/ec2_asg.py

```diff
@@ -1523,6 +1542,37 @@ def replace(connection):
     return changed, asg_properties
 
 
+def detach(connection):
+    group_name = module.params.get('name')
+    detach_instances = module.params.get('detach_instances')
+    as_group = describe_autoscaling_groups(connection, group_name)[0]
+    decrement_desired_capacity = module.params.get('decrement_desired_capacity')
+    min_size = module.params.get('min_size')
+    props = get_properties(as_group)
+    instances = props['instances']
+
+    # check if provided instance exists in asg
+    for instance_id in detach_instances:
+        if instance_id not in instances:
+            module.fail_json(msg="Provided instance ID: {0} does not belong to the AutoScalingGroup: {1}".format(
+                instance_id, group_name))
+
+    # check if minimum size is greater than what setting decrement_desired_capacity would make it
+    if min_size and min_size > len(instances) - len(detach_instances) and decrement_desired_capacity:
```

Comment:
Shouldn't be less than `min_size`?

Reply:
Hi @alinabuzachis , apologies for having a bit weird ```if``` statement, I'll make it more clear right away.

Also to explain this check, consider following example task from above playbook https://github.com/ansible-collections/community.aws/pull/933#issue-1129356479:
```
 # detach 2 instances from the asg and reduce the desired capacity from 3 to 1
    - name: detach 2 instance from the asg and reduce the desired capacity from 3 to 1
      community.aws.ec2_asg:
        name: test-asg-abc
        launch_config_name: my_new_lc
        health_check_period: 60
        health_check_type: ELB
        min_size: 3
        max_size: 5
        desired_capacity: 3
        region: us-east-2
        decrement_desired_capacity: true
        detach_instances:
          - '{{ instances_detach_replace[0] }}'
          - '{{ instances_detach_replace[1] }}'
```

Here, as the ```decrement_desired_capacity``` is set to ```true```, when these ```2``` instances are detached from the ASG, the `desired_capacity` is also decremented from `3` to `1` which will make `desired_capacity` less than the current set `min_size` in the `ASG`.
But as the `min_size` in the `ASG config` is set to `3` is causes the `boto3` to throw following error
```
"msg": "Failed to describe launch configurations: An error occurred (ValidationError) when calling the 
DetachInstances operation: AutoScalingGroup test-asg-xxx has min-size=3, max-size=5, and desired-size=3.
To detach 2 instances,  please update the AutoScalingGroup sizes appropriately.",
    "response_metadata": {
        "http_headers": {
            "connection": "close",
            "content-length": "409",
            "content-type": "text/xml",
            "date": "Tue, 15 Feb 2022 20:23:10 GMT",
            "x-amzn-requestid": "f7a74106-0bfd-44c4-83a9-0b8b27a8673f"
        },
        "http_status_code": 400,
        "request_id": "f7a74106-0bfd-44c4-83a9-0b8b27a8673f",
        "retry_attempts": 0
    }

```
To make this failures more easier to understand to the user I added the conditional check above which will throw this error in the same situation.
```
"msg": "Detaching instance(s) with 'decrement_desired_capacity' flag set reduces number of instances 
below min_size, please update AutoScalingGroup Sizes properly."
```

Reply:
Yes, I understood that. The thing confusing for me was the check. I would probably do something like this instead len(instances) - len(detach_instances) < min_size. But I mean, as you prefer.